addon-test-support/test.js

```diff
@@ -80,7 +80,9 @@ export default function test(testName, callback) {
           testTimeoutDeferred.resolve();
           resolve(data);
         } else {
-          setTimeout(poll, 10, resolve, reject);
+          setTimeout(function() {
```

Comment:
I think we need to use a fat arrow function (`() => {`) here, so that we maintain the proper lexical scope when we trigger the callback.

Reference: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout